tests/e2e/cucumber/steps/ui/session.ts

```diff
@@ -40,11 +41,22 @@ Given('{string} has logged out', LogOutUser)
 
 When('{string} logs out', LogOutUser)
 
+When('{string} fails to log in', async function (this: World, stepUser: string): Promise<void> {
```

Comment:
I think it makes sense to use `Then()` here, because this will always be an assertion.

```suggestion
Then('{string} fails to log in', async function (this: World, stepUser: string): Promise<void> {
```